fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -374,6 +374,26 @@ trait Parser[+T, ElemType, Repr] extends ParserResults[T, ElemType] with Precede
   }
 
   /**
+   * Extractor for pattern matching
+   *
+   *  For example:
+   *
+   *  {{{
+   *  val p1 = CharIn('a'to'z').! ~ CharIn('0'to'9').rep(1).!.map(_.toInt)
+   *  List("a42x") match {
+   *    case p1(x: String, y: Int) :: _ => // x is "a", y is 42
+   *  }
+   * }}}
+   */
+  def unapply[Repr <% IndexedSeq[ElemType]](input: Repr): Option[T] = {
```

Comment:
Sure. Having ```parse(Repr)``` instead of ```parse(IndexedSeq[ElemType]``` would make it even more simpler, with no ```<%```.

I am afraid it would confusing to use ```unapply``` with ```Iterator``` because one may write
```
input match {
  case parser1(x) =>
  case parser2(x) =>
}
```
reusing same input for another ```parse```.
Perhaps ```unapply``` could accept ```Iterable``` and create ```Iterator``` for each parse.